src/Microsoft.VisualStudio.ProjectSystem.Managed/Text/StringHash.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.VisualStudio.Text
+{
+    internal static class StableStringHash
+    {
+        public static int GetDeterministicHashCode(this string str)
+        {
+            unsafe
+            {
+                fixed (char* src = str)
+                {
+                    Contract.Assert(src[str.Length] == '\0', "src[this.Length] == '\\0'");
+                    Contract.Assert(((int)src) % 4 == 0, "Managed string should start at 4 bytes boundary");
```

Comment:
The code contracts can be removed as we don't use them.

Reply:
removed

done